src/Ewald.jl

```diff
@@ -408,17 +408,29 @@ struct DipoleRealCPU <: InteractionCPU
     int_mat :: OffsetArray{Mat3, 5, Array{Mat3, 5}}
 end
 
-function DipoleRealCPU(dip::DipoleDipole, crystal::Crystal, latsize)
+function DipoleRealCPU(dip::DipoleDipole, crystal::Crystal, latsize, sites_info::Vector{SiteInfo})
     @unpack strength, extent, η = dip
     lattice = Lattice(crystal, latsize)
-    return DipoleRealCPU(strength .* precompute_dipole_ewald(lattice; extent=extent, η=η))
+    A = strength .* precompute_dipole_ewald(lattice; extent=extent, η=η)
+    # Conjugate each matrix by the correct g matrices
+    for b1 in 1:nbasis(crystal)
+        S1, g1 = sites_info[b1].S, sites_info[b1].g
+        for b2 in 1:nbasis(crystal)
+            S2, g2 = sites_info[b2].S, sites_info[b2].g
+            for ijk in CartesianIndices(axes(A)[3:end])
+                A[b1, b2, ijk] = g1' * S1 * A[b1, b2, ijk] * S2 * g2
```

Comment:
This looks good. I guess the magnetic moment is:
`m = mu_B * S * g`,

with the `g` tensor going to the _right_ of `S`, and `mu_B` the Bohr magneton. In this case, the Bohr magneton can be effectively absorbed into `A`?

Reply:
Here, `S1` and `S2` are the scalar spin magnitudes, not the vector-spin values themselves. The idea is to absorb as much as possible into the dipole tensor.

You're right though that the Bohr magneton is still missing, and that it can also be absorbed into `A`. I'm planning to do this in a future "unit system" PR.

Reply:
I see. Might help to write:
`A[b1, b2, ijk] = (S1*S2) * g1' * A[b1, b2, ijk] * g2`
which to my mind emphasizes "`S1*S2` is a scalar prefactor".

Reply:
My intuition for writing it like this is that it's as if I just pulled out the inside of the expression (in parentheses):

`𝐒_i^T g^T A g 𝐒_j` = 
`𝐬_i^T (S_i g^T A g S_j) 𝐬_j`

in which case I should really swap the g and the S on both sides of the current code to be clear.

Do you prefer this (newly corrected) pulled-out-internals version, or the prefactor version? I have a moderate preference for the as-is with the g and S swapped, but am fine with either.